src/nativestore/NodeManager.cpp

```diff
@@ -25,13 +25,27 @@ limitations under the License.
 #include "PropertyLink.h"
 #include "RelationBlock.h"
 #include "iostream"
+#include <sys/stat.h>
 
 Logger node_manager_logger;
 pthread_mutex_t lockEdgeAdd;
 
+bool fileExists(const std::string& path) {
```

Comment:
Function for check file existance already there in Utils: https://github.com/miyurud/jasminegraph/blob/master/src/util/Utils.h#L69.
Use it

Reply:
But it doesn't check whether we have read permission for the file.

Reply:
But this is not a good method of checking it. 
You can check it by 
```cpp
access(fileName.c_str(), R_OK) == 0
```

Reply:
I think checking for read permission is not necessary as jasminegraph is anyway running in `root` mode

Reply:
@kandeeban15 Please let us know how you will resolve this.

Reply:
> I think checking for read permission is not necessary as jasminegraph is anyway running in `root` mode
@Ishad-M-I-M 
Why do we need to run JasmineGraph in root mode? So far we did not require it to run on root mode.

Reply:
It's running in docker, so it will have root privileges in container by default

Reply:
We will check for read permissions using a utility function defined in the Utils.cpp.